Some unit tests failing when run against Chrome

By default, our unit tests are all run against PhantomJS, because it's the only _easy_ way to run them headless.

The exception is when we do `grunt test:debug`. This launches a Chrome browser so we have access to all the debug tools.

We recently found that a few of our unit tests were _only_ failing when run with Chrome (I guess no one had run `grunt test:debug` recently against the full suite). Specifically

```
hrome 43.0.2357 (Mac OS X 10.10.3) rxFormItem textarea special case should put text-area-label class on the main div FAILED
	AssertionError: expected false to be true
Chrome 43.0.2357 (Mac OS X 10.10.3) rxFormUtils getSelectedOptionForTable returns the selected option for the table with id tableId FAILED
	AssertionError: expected undefined to deeply equal { rowId: 'table1_rowId1' }
	    at Context.<anonymous> (base/src/rxForm/rxForm.spec.js?eb992222af5bee3d1e04f8367e8fe4f0534c7b06:445:80)
Chrome 43.0.2357 (Mac OS X 10.10.3) rxFormUtils getSelectedOptionForTabSet returns the selected option for the tabset FAILED
	TypeError: Cannot read property 'rowId' of undefined
	    at Object.rxFormUtils.getSelectedOptionForTabSet (base/src/rxForm/rxForm.js?0d03ed1d97ecc3d5da22e552e530565d150a8130:328:72)
	    at Context.<anonymous> (base/src/rxForm/rxForm.spec.js?eb992222af5bee3d1e04f8367e8fe4f0534c7b06:459:35)
Chrome 43.0.2357 (Mac OS X 10.10.3) rxFormUtils getSelectedOptionForTabSet returns the selected option without the field id if no field-id exists FAILED
	TypeError: Cannot read property 'rowId' of undefined
	    at Object.rxFormUtils.getSelectedOptionForTabSet (base/src/rxForm/rxForm.js?0d03ed1d97ecc3d5da22e552e530565d150a8130:328:72)
	    at Context.<anonymous> (base/src/rxForm/rxForm.spec.js?eb992222af5bee3d1e04f8367e8fe4f0534c7b06:468:35)
Chrome 43.0.2357 (Mac OS X 10.10.3): Executed 635 of 635 (4 FAILED) (5.738 secs / 0.773 secs)
```

--- a/src/rxForm/rxForm.js
+++ b/src/rxForm/rxForm.js
@@ -45,7 +45,7 @@ angular.module('encore.ui.rxForm', ['ngSanitize', 'encore.ui.rxMisc'])
 
                 var inputField = el[0].querySelector(inputSelectors);
 
-                scope.isTextArea = _.has(inputField, 'type') && inputField.type === 'textarea';
+                scope.isTextArea = _.isObject(inputField) && inputField.type === 'textarea';
 
                 // make sure an input field is found
                 if (!_.isObject(inputField)) {
@@ -297,7 +297,7 @@ angular.module('encore.ui.rxForm', ['ngSanitize', 'encore.ui.rxMisc'])
  * getSelectedOptionForTabSet(tabsetId)
  * </pre>
  */
-.factory('rxFormUtils', function () {
+.factory('rxFormUtils', function ($document) {
 
     var rxFormUtils = {};
 
@@ -307,8 +307,9 @@ angular.module('encore.ui.rxForm', ['ngSanitize', 'encore.ui.rxMisc'])
     // @returns {object} The rowId of the selected option
     rxFormUtils.getSelectedOptionForTable = function (tableId) {
         var selectedRow;
-        var row = document.querySelector('rx-form-option-table#' + tableId + ' .selected input');
-        if (!_.isEmpty(row)) {
+        var row = $document[0].querySelector('rx-form-option-table#' + tableId + ' .selected input');
+
+        if (_.isObject(row) && 'value' in row) {
             selectedRow = { rowId: row.value };
         }
         return selectedRow;
@@ -321,7 +322,7 @@ angular.module('encore.ui.rxForm', ['ngSanitize', 'encore.ui.rxMisc'])
         var selectedOption;
         var xpathToTable = '//div[@id="' + tabsetId +
             '"]//tr[contains(@class, "selected")]//ancestor::rx-form-option-table';
-        var result = document.evaluate(xpathToTable, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null);
+        var result = $document[0].evaluate(xpathToTable, $document[0], null, XPathResult.FIRST_ORDERED_NODE_TYPE, null);
         if (result.singleNodeValue) {
             var table = result.singleNodeValue;
             var fieldId = table.getAttribute('field-id');

--- a/src/rxForm/rxForm.spec.js
+++ b/src/rxForm/rxForm.spec.js
@@ -389,15 +389,16 @@ describe('rxFormOptionTable (Radio)', function () {
 
 describe('rxFormUtils', function () {
 
-    var rxFormUtilsSvc, rootScope;
+    var rxFormUtilsSvc, rootScope, $document;
 
     var template;
 
     beforeEach(function () {
         module('encore.ui.rxForm');
 
-        inject(function ($rootScope, rxFormUtils) {
+        inject(function ($rootScope, _$document_, rxFormUtils) {
             rootScope = $rootScope;
+            $document = _$document_;
             rxFormUtilsSvc = rxFormUtils;
         });
 
@@ -433,7 +434,7 @@ describe('rxFormUtils', function () {
                      '</div></div></div>');
 
         rootScope.$digest();
-        $(document.body).append(template);
+        $($document[0].body).append(template);
     });
 
     describe('getSelectedOptionForTable', function () {

